File: mobile/src/main/java/com/alexstyl/specialdates/contact/ContactCache.java
package com.alexstyl.specialdates.contact;

import android.support.v4.util.LruCache;

class ContactCache<T extends Contact> {

    private final LruCache<Long, T> cache;

    ContactCache(int maxSize) {
        this.cache = new LruCache<>(maxSize);
    }

    void addContact(T contact) {
        cache.put(keyFor(contact), contact);
    }

    T getContact(Long id) {
        return cache.get(id);
    }

    private Long keyFor(T contact) {
        return contact.getContactID();
    }

    public int size() {
        return cache.size();
    }

    public void evictAll() {
        cache.evictAll();
    }

    public boolean containsContactWith(long contactID) {
        return cache.get(contactID) != null;
    }
}

File: memento/src/main/java/com/alexstyl/specialdates/contact/ContactCache.kt
package com.alexstyl.specialdates.contact

import android.LruCache


class ContactCache(maxSize: Int) {

    private val cache: LruCache<Long, Contact> = LruCache(maxSize)

    fun addContact(contact: Contact) {
        cache.put(keyFor(contact), contact)
    }

    fun getContact(id: Long): Contact? = cache.get(id)

    fun size(): Int = cache.size()

    fun evictAll() {
        cache.evictAll()
    }

    fun addContacts(contacts: List<Contact>) {
        contacts.forEach {
            cache.put(it.contactID, it)
        }
    }

    private fun keyFor(contact: Contact): Long = contact.contactID
}
